sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -94,214 +94,61 @@ export class EventHubSender extends LinkEntity {
-              "the sender link.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          logger.verbose(
-            "[%s] close() method of Sender '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the sender link.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
-        logger.verbose(
-          "[%s] close() method of Sender '%s' with address '%s' was not called. There " +
-            "was no accompanying error as well. This is a candidate for re-establishing " +
-            "the sender link.",
-          this._context.connectionId,
-          this.name,
-          this.address
-        );
-      } else {
-        const state: any = {
-          wasCloseInitiated: wasCloseInitiated,
-          senderError: senderError,
-          _sender: this._sender
-        };
-        logger.verbose(
-          "[%s] Something went wrong. State of sender '%s' with address '%s' is: %O",
-          this._context.connectionId,
-          this.name,
-          this.address,
-          state
-        );
-      }
-      if (shouldReopen) {
-        await defaultLock.acquire(this.senderLock, () => {
-          const options: AwaitableSenderOptions = this._createSenderOptions(
-            Constants.defaultOperationTimeoutInMs,
-            true
-          );
-          // shall retry forever at an interval of 15 seconds if the error is a retryable error
-          // else bail out when the error is not retryable or the oepration succeeds.
-          const config: RetryConfig<void> = {
-            operation: () => this._init(options),
-            connectionId: this._context.connectionId,
-            operationType: RetryOperationType.senderLink,
-            connectionHost: this._context.config.host,
-            retryOptions: {
-              maxRetries: Constants.defaultMaxRetriesForConnection,
-              retryDelayInMs: 15000
-            }
-          };
-          return retry<void>(config);
-        });
-      }
-    } catch (err) {
       logger.verbose(
-        "[%s] An error occurred while processing onDetached() of Sender '%s' with address " +
-          "'%s': %O",
+        "[%s] 'session_close' event occurred on the session of sender '%s' with address '%s'. " +
+          "Value for isSessionItselfClosed on the session is: '%s'",
         this._context.connectionId,
         this.name,
         this.address,
-        err
+        sender ? sender.isSessionItselfClosed().toString() : undefined
       );
-    }
+      if (sender && !this.isConnecting) {
+        // Call close to clean up timers & other resources
+        await sender.close();
```

Comment:
Same comment here about errors on close.

Reply:
Fixed with https://github.com/Azure/azure-sdk-for-js/pull/6737/commits/06e9251b0941110ccdaa398a637d4fba818f69f1